HIP/hip-0000-schedule-service-system-contract.md

```diff
@@ -0,0 +1,108 @@
+
+## User stories
+
+1. As an EOA I would like to initiate a smart contract transaction that schedules a supported transaction.
+2. As an EOA I would like to initiate a smart contract transaction that allows me to sign a scheduled transaction.
+3. As an EOA I would like to initiate a smart contract transaction that prompts a contract to authorize a scheduled transaction.
+4. As an EOA I would like to initiate a smart contract transaction that allows me to extract information about a scheduled transaction.
+  
+## Specification
+
+The ledger HSCS will utilize the existing scheduled transaction service supported on the ledger within the system contract logic. 
+
+To achieve this a new Hedera Schedule Service (HSS) system contract will need to be created to encompass and expose the necessary scheduled transaction features.
+
+### Hedera Schedule Service (HSS) system contract
+
+A new `IHederaScheduleService` interface will be implemented to allow accounts to authorize a pre-existing scheduled transaction via smart contracts
+
+|   Hash        |   Selector                                                                                        |   In contract support |
+|---------------|---------------------------------------------------------------------------------------------------|-----------------------|
+| `0xf0637961`  | `authorizeSchedule(address) external returns (int64 responseCode)`                                | Y                     |
+| `0x5e147101`  | `getScheduledTransactionInfo(address) external returns (bytes memory transactionProtobufBytes)`   | Y                     |
+| `0xd797b304`  | `signSchedule(address) external returns (int64 responseCode)`                                     | N                     |
+| `0x358eeb03`  | `signSchedule(address, bytes) external returns (int64 responseCode)`                              | Y                     |
+
+Since `signSchedule(address scheduleAddress) returns (int64 responseCode)` relies on an implicit signature it will only callable by EOA’s via the IHRC facade.
+In this case the signature will be the inner ECDSA signature found in the RLP encoded `EthereumTransaction`. 
+For `Contract` and `ContractCreate`  any applicable signature found in the signature map will be utilized as in `ScheduleSign`.
+
+Note this HIP does not provide an API to create a scheduled transaction. This is left to future HIPs to present the appropriate transactions that may be scheduled.
+No further protobuf or application level changes are needed as HSS is already implemented and functional.
+
+## Backwards Compatibility
+
+Backwards compatibility is ensured as no existing features are modified. Similar to HTS system contract this HIP simply exposes HAPI entity functionality and the system comtract will utilize the same HAPI services on the node.
+
+## Security Implications
+
+Existing security consideration such as throttles will remain applicable. 
+Additional considerations may include
```

Comment:
Finish or remove.

Reply:
The rest of the section are the completion of the section. Let me update it to be more explicit

Reply:
Updated